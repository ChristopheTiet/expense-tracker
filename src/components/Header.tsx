import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1 className="text-xl p-4">Expense Tracker</h1>
    <div className="space-x-3 p-4">
      <NavLink
        to="/"
        className={({ isActive }) => (isActive ? "text-red-500" : "text-black")}
      >
        Dashboard
      </NavLink>
      <NavLink
        to="create"
        className={({ isActive }) => (isActive ? "text-red-500" : "text-black")}
      >
        Add Expense
      </NavLink>
      <NavLink
        to="edit"
        className={({ isActive }) => (isActive ? "text-red-500" : "text-black")}
      >
        Edit Expense
      </NavLink>
      <NavLink
        to="help"
        className={({ isActive }) => (isActive ? "text-red-500" : "text-black")}
      >
        Help
      </NavLink>
    </div>
  </header>
);

export default Header;
