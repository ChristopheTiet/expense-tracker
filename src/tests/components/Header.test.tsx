import renderer from "react-test-renderer";
import Header from "../../components/Header";
import { BrowserRouter } from "react-router-dom";

test("should render Header correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
