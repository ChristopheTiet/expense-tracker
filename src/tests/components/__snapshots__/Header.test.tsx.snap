// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1
    className="text-xl p-4"
  >
    Expense Tracker
  </h1>
  <div
    className="space-x-3 p-4"
  >
    <a
      aria-current="page"
      className="text-red-500"
      href="/"
      onClick={[Function]}
    >
      Dashboard
    </a>
    <a
      className="text-black"
      href="/create"
      onClick={[Function]}
    >
      Add Expense
    </a>
    <a
      className="text-black"
      href="/edit"
      onClick={[Function]}
    >
      Edit Expense
    </a>
    <a
      className="text-black"
      href="/help"
      onClick={[Function]}
    >
      Help
    </a>
  </div>
</header>
`;
